canal/config.go

```diff
@@ -34,6 +34,9 @@ type DumpConfig struct {
 	// 'FLUSH TABLES WITH READ LOCK'
 	SkipMasterData bool `toml:"skip_master_data"`
 
+	// set --set-gtid-purged none, auto; none for gtid is disabled or "version too low", auto for gtid_mode=on;
```

Comment:
why when set to `auto` then enable gtid_mode? how about use `on` or `off`, or use bool type

Reply:
I think on or off might be misunderstanding. mysqldump of Mariadb is not compatible with that of MySQL. 
If off is set, it look like --set-gtid-purged=off might be set, but Mariadb is not supported.
So i would prefer to use auto of MySQL, none for Mariadb.
What do you think ?

Reply:
> set_gtid_purged: 'on', 'auto' or 'none' can be set

In the above error message, seem 'on' can be set, but here we only need to set 'none' or 'auto', no 'on'?

Reply:
yes, https://github.com/siddontang/go-mysql/pull/444#discussion_r341884561
canal/canal.go should be fixed. Could you fix it ? Or I fix it by myself. 
Sorry about these trouble